timelock-api/src/test/java/com/palantir/atlasdb/timelock/api/ConjureLockRequestTest.java

```diff
@@ -0,0 +1,152 @@
+
+    // AtlasDB (Client) serializes and TimeLock (Server) deserializes ConjureLockRequest objects.
+    // These are the respective mappers used internally by Conjure.
+    private static final ObjectMapper SERIALIZATION_MAPPER =
+            ObjectMappers.newClientObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+    private static final ObjectMapper DESERIALIZATION_MAPPER =
+            ObjectMappers.newServerObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+    // This mapper is used to ensure that two JSONs are equal excluding indentation
+    private static final ObjectMapper VERIFYING_MAPPER = new ObjectMapper();
+
+    @Test
+    public void oldConjureLockRequestSerializesDeserializes() {
+        assertSerializedEquals(OLD_CONJURE_LOCK_REQUEST, OLD_CONJURE_LOCK_REQUEST_FILE);
+        assertDeserializedEquals(OLD_CONJURE_LOCK_REQUEST_FILE, OLD_CONJURE_LOCK_REQUEST, OldConjureLockRequest.class);
+    }
+
+    // Requires excludeEmptyOptionals = true
+    @Test
+    public void absentMetadataSerializesDeserializes() {
+        assertSerializedEquals(DEFAULT_CONJURE_LOCK_REQUEST, OLD_CONJURE_LOCK_REQUEST_FILE);
+        assertDeserializedEquals(OLD_CONJURE_LOCK_REQUEST_FILE, DEFAULT_CONJURE_LOCK_REQUEST, ConjureLockRequest.class);
+    }
+
+    @Test
+    public void presentMetadataSerializesDeserializes() {
+        ConjureLockRequest requestWithMetadata = ConjureLockRequest.builder()
+                .from(DEFAULT_CONJURE_LOCK_REQUEST)
+                .metadata(CONJURE_LOCK_REQUEST_METADATA)
+                .build();
+        assertSerializedEquals(requestWithMetadata, WITH_METADATA_FILE);
+        assertDeserializedEquals(WITH_METADATA_FILE, requestWithMetadata, ConjureLockRequest.class);
+    }
+
+    @Test
+    public void newServerCanHandleMissingMetadata() {
+        assertDeserializedEquals(OLD_CONJURE_LOCK_REQUEST_FILE, DEFAULT_CONJURE_LOCK_REQUEST, ConjureLockRequest.class);
+    }
+
+    // Possible because we do not use strict objects
+    @Test
+    public void oldServerCanHandlePresentMetadata() {
+        assertDeserializedEquals(WITH_METADATA_FILE, OLD_CONJURE_LOCK_REQUEST, OldConjureLockRequest.class);
+    }
+
+    private static <T> void assertSerializedEquals(T object, String jsonFileName) {
+        try {
+            Path path = getJsonPath(jsonFileName);
+            if (REWRITE_JSON_BLOBS) {
+                SERIALIZATION_MAPPER.writeValue(path.toFile(), object);
+            }
+            String serialized = SERIALIZATION_MAPPER.writeValueAsString(object);
+            assertThat(VERIFYING_MAPPER.readTree(serialized))
+                    .as("Serialization yields semantically identical JSON representation")
+                    .isEqualTo(VERIFYING_MAPPER.readTree(Files.readString(path)));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static <T> void assertDeserializedEquals(String jsonFileName, T object, Class<T> clazz) {
```

Comment:
also only other style nit, is that we have arguments reversed from `assertSerializedEquals`. I'd expect `assertDeserializedEquals(T object, String jsonFileName, Class<T> clazz)`

Reply:
This was deliberate to maintain the convention that we do `assertThat(actual).isEqualTo(expected)`. The deserialized JSON object is the `actual` value and the object is the `expected` value.

Reply:
ah, then that makes sense!